File: library/src/main/java/androidz/AndroidzInitializer.java
package androidz;

import android.content.Context;

import java.util.Collections;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.startup.Initializer;


public class AndroidzInitializer implements Initializer<AndroidzInitializer> {

    @NonNull
    @Override
    public AndroidzInitializer create(@NonNull Context context) {
        Androidz.initialize(context);
        return this;
    }

    @NonNull
    @Override
    public List<Class<? extends Initializer<?>>> dependencies() {
        return Collections.emptyList();
    }
}

File: library/src/main/java/androidz/AndroidzInitializer.kt
package androidz

import android.content.Context
import androidx.startup.Initializer


class AndroidzInitializer : Initializer<AndroidzInitializer> {

    override fun create(context: Context): AndroidzInitializer {
        Androidz.initialize(context)
        return this
    }

    override fun dependencies(): List<Class<out Initializer<*>>> {
        return emptyList()
    }
}
